tests/run/unicode_identifiers.pyx

```diff
@@ -1,9 +1,48 @@
 # -*- coding: utf-8 -*-
 # cython: language_level=3
-# mode: compile
+# mode: run
 # tag: pep3131
 
-"ναμε1"
+# Code with unicode_identifiers can be compiled with Cython running either Python 2 or 3.
+# However it can only be run in Python 2. Therefore no doctests are run in Python2.
```

Comment:
-> `can only be run in Python 3`

That's not entirely true, though. A compiled ASCII-named Python function that internally uses and calls non-ASCII names and functions should be usable from a Python doctest. Would be nice to have this case included (in both Py2 and Py3). If that works, it allows Cython modules also in Python 2 that can use Unicode identifiers internally, as long as they provide an ASCII-only Python-level API.

Reply:
I've added a few runtime tests for using Python 2 as you described. I'd be reluctant to really advertise this as a feature since it's hard to avoid any unicode leaking back to Python when it shouldn't do and there's no attempt to diagnose when the "ASCII-only Python-level API" is crossed, but practically a lot of it works.

The tests are fairly basic since it's surprisingly hard to write doctests that output unicode and work in all versions (I gave up on inspecting the traceback and locals in Python 2 for example).